spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/support/SimpleSpannerRepository.java

```diff
@@ -154,8 +157,36 @@ public Page findAll(Pageable pageable) {
 			}
 		}
 		else {
-			k = key instanceof Key ? (Key) key : Key.of(key);
+			k = Key.class.isAssignableFrom(key.getClass()) ? (Key) key
+					: Key.of(convertKeyPart(key));
 		}
 		return operation.apply(k);
 	}
+
+	private Object convertKeyPart(Object object) {
+		if (MappingSpannerConverter
+				.isValidSpannerKeyType(ConversionUtils.boxIfNeeded(object.getClass()))) {
```

Comment:
`isValidSpannerKeyType` should be non-static, even if the logic currently depends on a set of statically predefined values. Introducing these hard-coded static depedencies makes the code hard to test and extend later, it's too rigid of a design with no particular benefit.